indexd/src/app_client.rs

```diff
@@ -43,63 +96,207 @@ pub struct SlabPinParams {
 }
 
 impl Client {
-    #[allow(dead_code)]
-    pub fn new(url: String, password: Option<String>) -> Self {
-        Self {
+    pub fn new<U: IntoUrl>(url: U, app_key: PrivateKey) -> Result<Self> {
+        Ok(Self {
             client: reqwest::Client::new(),
-            url: url.trim_end_matches('/').to_string(),
-            password,
+            url: url.into_url()?,
+            app_key,
+        })
+    }
+
+    /// Checks if the application is authenticated with the indexer. It returns
+    /// true if authenticated, false if not, and an error if the request fails.
+    pub async fn check_app_authenticated(&self, status_url: &Url) -> Result<bool> {
+        let resp = self
```

Comment:
What does this correspond to on the App API? I'm seeing

`GET /auth/check` and `GET /auth/connect/:requestID/status`. No post endpoint for checking auth